### ISSUE TITLE
repository name filter support

### ISSUE THREAD

**Issue by shizhMSFT** [Azure/acr-cli#57]
**What is the problem you're trying to solve**
Request new feature: support regex filtering on repository names.

**Describe the solution you'd like**
Making the following command possible:
```
acr purge \
    --registry myacr \
    --filter "test[0-9]*:.*"
    --ago 30d
```

**Comment by jemag** [comment]
This would be great. 

Currently, unless I am mistaken, there does not seem to be a way to purge all images from all repositories older than X days? 

I would need to create 1 task per repository, which would quickly become cumbersome with 100+ repositories, especially if you only need to apply the same purge criteria (e.g.: days ago) on all of them. For that specific use case, another option would be to not set the --filter as mandatory, and when filter is not provided, the purge will be applied to all repositories according to the other criteria passed (e.g.: --ago)

**Comment by AKTheKnight** [comment]
Yes please. There's currently 20 repositories within one of our registries and I don't want to chain --filter or setup a task for all of them!

Either regex in the --filter or no requirement for the --filter flag would be great

**Comment by matt-sdc** [comment]
Kinda like @AKTheKnight  said, I'd like to be able to run the command on all repos, my case would be something like, --filter "*:.*" which should just be simplied to no filter

**Comment by rlees85** [comment]
Another vote for this functionality that I am surprised is not present as standard. When creating ACR using automation (like Terraform) this is the ideal time to also create the ACR task for house keeping. Repositories can be created ad-hoc by anyone running "docker build" so there is no way to automatically insert a task to do this.

Being able to just purge against the entire ACR would be awesome. I tried to create the functionality by using an azure-cli container with `acr` command included but it doesn't work since az-cli cannot authenticate.

**Comment by butzist** [comment]
Until this is merged, the image is also available as docker.io/datahow/acr:latest

**Comment by Wwwsylvia** [comment]
Thanks @butzist for contributing this feature! We have merged #88 .
